wasm/jit/jit_arm64.go

```diff
@@ -554,8 +624,229 @@ func (c *arm64Compiler) compileBrTable(o *wazeroir.OperationBrTable) error {
+
+	// Finally, we are ready to write the address of the target function's *compiledFunction into the new callframe.
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		compiledFunctionAddressRegister,
+		callFrameStackTopAddressRegister, callFrameCompiledFunctionOffset)
+
+	// 4) Set ra.current so that we can return back to this function properly.
+	//
+	// First, Get the return address into the tmpRegisters[2].
+	c.readInstructionAddress(obj.AJMP, tmpRegisters[2])
+	// Then write the address into the callframe.
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		tmpRegisters[2],
+		// "ra.current" is BELOW the top address. See the above example for detail.
+		callFrameStackTopAddressRegister, -(callFrameDataSize - callFrameReturnAddressOffset),
+	)
+
+	// Everthing is done to make function call now.
+	// We increment the callframe stack pointer.
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		reservedRegisterForEngine, engineGlobalContextCallFrameStackPointerOffset,
+		tmpRegisters[2])
+	c.applyConstToRegisterInstruction(arm64.AADD, 1, tmpRegisters[2])
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		tmpRegisters[2],
+		reservedRegisterForEngine, engineGlobalContextCallFrameStackPointerOffset)
+
+	// Then, br into the target function's initial address.
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		compiledFunctionAddressRegister, compiledFunctionCodeInitialAddressOffset,
+		tmpRegisters[2])
+	c.emitUnconditionalBranchToAddressOnRegister(tmpRegisters[2])
+
+	// All the registers used are temporary so we mark them unused.
+	c.markRegisterUnused(tmpRegisters...)
+
+	// We consumed the function parameters from the stack after call.
+	for i := 0; i < len(functype.Params); i++ {
+		c.locationStack.pop()
+	}
+
+	// Also, the function results were pushed by the call.
+	for _, t := range functype.Results {
+		loc := c.locationStack.pushValueLocationOnStack()
+		switch t {
+		case wasm.ValueTypeI32, wasm.ValueTypeI64:
+			loc.setRegisterType(generalPurposeRegisterTypeInt)
+		case wasm.ValueTypeF32, wasm.ValueTypeF64:
+			loc.setRegisterType(generalPurposeRegisterTypeFloat)
+		}
+	}
+
+	// On the function return, we initialize the state for this function.
+	c.initializeReservedStackBasePointerRegister()
+
+	// TODO: initialize module context, and memory pointer.
+	return nil
+}
+
+// readInstructionAddress adds an ADR instruction to set the absolute address of "target instruction"
+// into destinationRegister. "target instruction" is specified by beforeTargetInst argument and
+// the target is determined by "the instruction right after beforeTargetInst type".
+//
+// For example, if beforeTargetInst == RET and we have the instruction sequence like
+// ADR -> X -> Y -> ... -> RET -> MOV, then the ADR instruction emitted by this function set the absolute
+// address of MOV instruction into the destination register.
+func (c *arm64Compiler) readInstructionAddress(beforeTargetInst obj.As, destinationRegister int16) {
```

Comment:
should this be compileXXX?

Reply:
nope, this is arm64 specific and don't need to be exposed as compiler interface

Reply:
well we add the XX implements XXX when it is part of the interface, we can still use a common prefix when there are functions that do the same sort of things. That was actually part of why I suggested we enforce the interface in the first place, so that we can tell the difference between naming conventions vs what are common across

Reply:
no I don't think so, this could be re-written and removed in anyway after #221 and I don't think we need this for other archs (e.g. RISC V).  Having methods on interface which are not used anywhere seems weird to me, so let me leave it as-is.

Reply:
you misunderstood. but the outcome is fine. What I meant is that naming convention != interface. we can re-use a naming convention for things that add instructions without adding them to the interface. put another way no reason for us to require adding an interface just to prefix "compileX".

Reply:
ah ok got it.

Reply:
so then in what way `compile*` (or whatever prefix) common prefix convention helps here?

Reply:
example would help

Reply:
compileXXX is used currently for operations that end up adding instructions, effecting state via `c.addInstruction(inst)`. I don't see a point using different prefixes to begin describing things that end up doing that. I have been looking at things and if I see compileXXX I think first this is going to add instructions, then I look at what it is doing if it matches the intent either via inherited doc from the interface or what's on the non-shared thing. I look to see if there's an "addInstruction" as formerly "compile" was a keyword for me to look for state changes vs ancillary functions that do not generate instructions. Does this make sense?

Reply:
yeah make sense esp this

> effecting state via c.addInstruction(inst).